test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/Utility/NuGetSearchServiceReconnectorTests.cs

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.ServiceHub.Framework;
+using Microsoft.VisualStudio.Threading;
+using Moq;
+using NuGet.PackageManagement.UI.Utility;
+using NuGet.VisualStudio.Internal.Contracts;
+using Xunit;
+
+namespace NuGet.PackageManagement.UI.Test.Utility
+{
+    public class NuGetSearchServiceReconnectorTests
+    {
+        private readonly JoinableTaskFactory _jtf;
+
+        public NuGetSearchServiceReconnectorTests()
+        {
+#pragma warning disable VSSDK005 // Avoid instantiating JoinableTaskContext
+            var jtc = new JoinableTaskContext();
+#pragma warning restore VSSDK005 // Avoid instantiating JoinableTaskContext
+            _jtf = new JoinableTaskFactory(jtc);
+        }
+
+        [Fact]
+        public async Task CreateAsync_NullArguments_ThrowsArgumentNullException()
+        {
+            // Arrange
+            var serviceBroker = new Mock<IServiceBroker>();
+
+            // Act & Assert
+            ArgumentNullException exception = await Assert.ThrowsAsync<ArgumentNullException>(() => NuGetSearchServiceReconnector.CreateAsync(serviceBroker: null, _jtf, CancellationToken.None));
+            Assert.Equal("serviceBroker", exception.ParamName);
+
+            exception = await Assert.ThrowsAsync<ArgumentNullException>(() => NuGetSearchServiceReconnector.CreateAsync(serviceBroker.Object, jtf: null, CancellationToken.None));
+            Assert.Equal("jtf", exception.ParamName);
+        }
+
+        [Fact]
+        public async Task Object_NormalCreation_GetInstanceFromServiceBroker()
+        {
+            // Arrange
+            var searchService = new Mock<INuGetSearchService>();
+
+            var serviceBroker = new Mock<IServiceBroker>();
+            serviceBroker.Setup(sb =>
+#pragma warning disable ISB001 // Dispose of proxies
+            sb.GetProxyAsync<INuGetSearchService>(It.IsAny<ServiceJsonRpcDescriptor>(), It.IsAny<ServiceActivationOptions>(), It.IsAny<CancellationToken>())
+#pragma warning restore ISB001 // Dispose of proxies
+            )
+                .ReturnsAsync(searchService.Object);
+
+            var target = await NuGetSearchServiceReconnector.CreateAsync(serviceBroker.Object, _jtf, CancellationToken.None);
```

Comment:
explicit type

Reply:
.NET doesn't support async constructors, hence why the usual pattern is to have an static async create method. Is it not clear enough?

We might as well use C# 9's `Type name = new();` feature, and completely forbid `var`.

Reply:
Looks like you went with `NuGetSearchServiceReconnector target` which meets our Coding Guideline # 10.